rtree/core.py

```diff
@@ -69,6 +70,30 @@ def free_error_msg_ptr(result, func, cargs):
     rt.Index_Free(p)
     return retvalue
 
+# Read back env vars that were set up during package installation.
+try:
+    for line in pkgutil.get_data('rtree', 'ENVIRON.txt').decode().split():
+        # Ignore lines with wrong number of '=' signs
+        kv = line.split('=')
+        if len(kv) != 2:
+            continue
+        key, val = kv
+
+        # Only allow certain vars to be set
+        legal_envs = ('SPATIALINDEX_LIBRARY', 'SPATIALINDEX_C_LIBRARY')
+        if key not in legal_envs:
+            raise ValueError(
+                'Illegal variable in ENVIRON.txt: %s.  Must be one of %s' %
+                (key, str(legal_envs)))
+
+        # Only set from config file if not already set in env
+        if key not in os.environ:
+            os.environ[key] = val
```

Comment:
Yes, this gets the precedence right, but instead of using `os.environ` as data structure in this module (bc it's a global) can you implement something like the following instead?

```
libname = 'spatialindex_c.dll'
if 'SPATIALINDEX_C_LIBRARY' in config_file_dict:
    libname = config_file_dict['SPATIALINDEX_C_LIBRARY']
if 'SPATIALINDEX_C_LIBRARY' in os.environ:
    libname = os.environ['SPATIALINDEX_C_LIBRARY']

lib_path, lib_name = os.path.split(libname)
rt = _load_library(lib_name, ctypes.cdll.LoadLibrary, (lib_path,))
```

It's better if we don't update the environment unnecessarily and only access it once.